sdk/resourcemanager/azure-resourcemanager-resources/src/main/java/com/azure/resourcemanager/resources/fluentcore/utils/SdkContext.java

```diff
@@ -3,44 +3,46 @@
 
 package com.azure.resourcemanager.resources.fluentcore.utils;
 
+import com.azure.core.management.provider.DelayProvider;
+import com.azure.core.management.provider.IdentifierProvider;
 import reactor.core.scheduler.Scheduler;
 import reactor.core.scheduler.Schedulers;
 
-import java.io.File;
-import java.io.IOException;
 import java.time.Duration;
 import java.time.OffsetDateTime;
+import java.util.function.Function;
 
 /**
  * The class to contain the common factory methods required for SDK framework.
  */
 public class SdkContext {
-    private ResourceNamerFactory resourceNamerFactory = new ResourceNamerFactory();
-    private static DelayProvider delayProvider = new DelayProvider();
-    private static FileProvider fileProvider = new FileProvider();
+    private Function<String, IdentifierProvider> identifierFunction = name -> new ResourceNamer(name);
```

Comment:
Use function here so in test mode we can always use the testResourceNamer from azure-core-test by changing the function.